### DISCUSSION TITLE
how to modify  weight NDarray which is not writable?

### DISCUSSION THREAD

**Discussion by zhaoningning** [apache/incubator-mxnet#7880] [Q&A]
hello guys:
          i'm writing a layer using mxnet CustomOp api, and  in this layer,i want to modify weights data which is not writable( throw a error). Is there a way to solve this problem?  Mxnet's  optimizer  certainly can update the weights , how does it work?

**Comment by eric-haibin-lin** [comment]
@piiswrong is there a way to declare inputs are mutated with custom op?

**Comment by szha** [comment]
@apache/mxnet-committers: This issue has been inactive for the past 90 days. It has no label and needs triage.

For general "how-to" questions, our [user forum](https://discuss.mxnet.io/) (and [Chinese version](https://discuss.gluon.ai/)) is a good place to get help.
